rust/src/storage/utils.rs

```diff
@@ -91,8 +92,12 @@ impl TryFrom<&Add> for ObjectMeta {
             Utc,
         );
         Ok(Self {
-            // TODO this won't work for absoute paths, since Paths are always relative to store.
-            location: Path::from(value.path.as_str()),
+            // TODO this won't work for absolute paths, since Paths are always relative to store.
+            location: Path::from_url_path(
+                utf8_percent_encode(value.path.as_str(), percent_encoding::NON_ALPHANUMERIC)
+                    .to_string()
+                    .as_str(),
+            )?,
```

Comment:
How about:

```suggestion
            location: Path::parse(value.path.as_str())?,
```

I've tested this works locally.